### ISSUE TITLE
How it is made in vanced?

### ISSUE THREAD

**Issue by ssSFrankSss** [PierfrancescoSoffritti/android-youtube-player#707]
Hi, I have downloaded Vanced app from playstore. And I wonder how they were able to fully customize the player and remove any YouTube logos, and include their own buttons? And also how is this legal with 1 million downloads - with Google letting this app to be on playstore :/

This question comes as I have read that removing YouTube logos is impossible. Can somebody point me to the right direction?

Thank you for this amazing library anyway :)

**Comment by PierfrancescoSoffritti** [comment]
Are you talking about Vanced Tube - Video Player? I agree, not sure that app will be there for long. It has beep published for the first time late last year, so not long ago. In my experience Google can be pretty slow in finding app's that break ToS.

**Comment by ssSFrankSss** [comment]
Yes! And now they released also an app called "Pure Tuber". I can not understand how they were able to remove all the youtube logos and fully customize player. I believe this is a webview and they were able somehow to remove all logos with HTML5 maybe who knows...

**Comment by PierfrancescoSoffritti** [comment]
Well it's possible to hack the web player to remove parts of it, see #329. It's not a good solution though, since it will break each time YouTube makes some changes to their code.

Still YouTube won't be happy with it, as you're breaking ToS.

**Comment by PierfrancescoSoffritti** [comment]
I just looked at Pure Tuber, can say almost for sure that it will be removed, since it's just a YouTube clone and break every possible rule. Unless something major changed in the ToS.

**Comment by ssSFrankSss** [comment]
Even though I am not good at HTML world this is a site where you could create your own custom player: -> https://empora.com/
They have found ways to remove everything, and change the style of the player. Then you can insert it with a webview. This is the only nice way I can think of right now.

**Comment by wadelzubair** [comment]
Pure Tuber its no loner available :

**Comment by wadelzubair** [comment]
Hi Sir @PierfrancescoSoffritti can you please take a look [here ](https://github.com/PierfrancescoSoffritti/android-youtube-player/issues/712)
